src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -3308,6 +3385,50 @@ GetRunnableBackgroundTask(void)
 }
 
 
+/*
+ * GetTaskTokens
+ * Checks whether the node ids in the task tokens are available
+ * If at least one of them is not available, it returns false.
+ * If yes, it takes all the needed tokens and returns true.
+ */
+static bool
+GetTaskTokens(HTAB *ParallelMovesPerNode, BackgroundTask *task)
+{
+	if (task->nodeTokens)
+	{
+		int node_token;
+
+		/* first check if all tokens are available*/
+		foreach_int(node_token, task->nodeTokens)
+		{
+			bool found;
+			ParallelMovesPerNodeEntry *hashEntry = hash_search(
+				ParallelMovesPerNode, &(node_token), HASH_ENTER, &found);
+			if (!found)
+			{
+				hashEntry->counter = 0;
+			}
+			else if (hashEntry->counter == MaxParallelMovesPerNode)
+			{
+				/* at least one token (this one) is not available */
+				return false;
+			}
+		}
+
+		/* then, take all needed tokens */
+		foreach_int(node_token, task->nodeTokens)
+		{
+			ParallelMovesPerNodeEntry *hashEntry = hash_search(
+				ParallelMovesPerNode, &(node_token), HASH_FIND, NULL);
+			Assert(hashEntry);
+			hashEntry->counter += 1;
```

Comment:
Token buckets in general start with a positive value and you take a token by subtracting (just like semaphores, but not necessarily thread safe). The approach you're doing clearly works too and we don't necessarily have to flip it. But I do think the code should align with the terminology used in the comments. So either we:
1. decrease from a positive counter and use the token bucket terminology in comments
2. increase from a 0 starting point and use different terminology, e.g. inflight tasks with a limit

I actually think I prefer the increase approach that you have now, because I think it automatically works as we want when changing MaxParallelMovesPerNode.